### ISSUE TITLE
Custom output for tests that use `isempty`

### ISSUE THREAD

**Issue by oxinabox** [JuliaLang/julia#30651]
If would be good if
writing 

```
foo = [1,2,3]
@test isempty(foo)
```

resulted in a nicely formatted message
like 
`Test failed, expected foo to be empy but it contains [1,2,3]`

or something to that effect.

You can work around this by saying instead `@test [] == foo` or similar.

This would be particularly nice for `isempty(detect_ambiguities(Base, Core))`.

Related: 
 - https://github.com/JuliaLang/julia/issues/26882
 - https://github.com/JuliaLang/julia/issues/25486
 - https://github.com/JuliaLang/julia/issues/25487

**Comment by omus** [comment]
I would also like this. I think the formatting should match the existing printing that is in place and look something like:
```julia
julia> using Test

julia> x = [1]
1-element Array{Int64,1}:
 1

julia> @test isempty(x)
Test Failed at REPL[3]:1
  Expression: isempty(x)
   Evaluated: isempty([1])
ERROR: There was an error during testing
```

**Comment by omus** [comment]
Addressed by #30721
